remote-source/main.go

```diff
@@ -0,0 +1,277 @@
+
+	// Run a refresh on the platform stack
+	_, err = refreshStack(ctx, platformStack, env.PlatformProject.Name)
+	if err != nil {
+		fmt.Printf("Error encountered refreshing %s stack: %v\n", env.PlatformProject.Name, err)
+		os.Exit(1)
+	}
+
+	// Run an update of the platform stack
+	_, err = updateStack(ctx, platformStack, env.PlatformProject.Name)
+	if err != nil {
+		fmt.Printf("Error encountered updating %s stack: %v\n", env.PlatformProject.Name, err)
+		os.Exit(1)
+	}
+
+	// Set config values for the data stack
+	dataStack.SetConfig(ctx, "aws:region", auto.ConfigValue{Value: env.Region, Secret: false})
+	dataStack.SetConfig(ctx, "baseOrgName", auto.ConfigValue{Value: env.Organization, Secret: false})
+	dataStack.SetConfig(ctx, "baseProjName", auto.ConfigValue{Value: env.BaseProject.Name, Secret: false})
+	dataStack.SetConfig(ctx, "baseStackName", auto.ConfigValue{Value: env.StackName, Secret: false})
+
+	// Run a refresh on the data stack
+	_, err = refreshStack(ctx, dataStack, env.DataProject.Name)
+	if err != nil {
+		fmt.Printf("Error encountered refreshing %s stack: %v\n", env.DataProject.Name, err)
+		os.Exit(1)
+	}
+
+	// Run an update of the data stack
+	_, err = updateStack(ctx, dataStack, env.DataProject.Name)
+	if err != nil {
+		fmt.Printf("Error encountered updating %s stack: %v\n", env.DataProject.Name, err)
+		os.Exit(1)
+	}
+
+	// Set config values for app stack
+	appStack.SetConfig(ctx, "platformOrgName", auto.ConfigValue{Value: env.Organization, Secret: false})
+	appStack.SetConfig(ctx, "dataOrgName", auto.ConfigValue{Value: env.Organization, Secret: false})
+	appStack.SetConfig(ctx, "platformProjName", auto.ConfigValue{Value: env.PlatformProject.Name, Secret: false})
+	appStack.SetConfig(ctx, "dataProjName", auto.ConfigValue{Value: env.DataProject.Name, Secret: false})
+	appStack.SetConfig(ctx, "platformStackName", auto.ConfigValue{Value: env.StackName, Secret: false})
+	appStack.SetConfig(ctx, "dataStackName", auto.ConfigValue{Value: env.StackName, Secret: false})
+
+	// Run a refresh on the app stack
+	_, err = refreshStack(ctx, appStack, env.AppProject.Name)
+	if err != nil {
+		fmt.Printf("Error encountered refreshing %s stack: %v\n", env.AppProject.Name, err)
+		os.Exit(1)
+	}
+
+	// Run an update of the app stack
+	_, err = updateStack(ctx, appStack, env.AppProject.Name)
+	if err != nil {
+		fmt.Printf("Error encountered updating %s stack: %v\n", env.AppProject.Name, err)
+		os.Exit(1)
+	}
+}
+
+func refreshStack(ctx context.Context, s auto.Stack, n string) (auto.RefreshResult, error) {
+	var res auto.RefreshResult
+	fmt.Printf("Starting refresh of %s stack\n", n)
+	tmp, err := os.CreateTemp(os.TempDir(), "")
+	if err != nil {
+		fmt.Printf("Error creating temporary file: %v\n", err)
+		return res, err
+	}
+	progressStreams := []io.Writer{os.Stdout, tmp}
+	res, err = s.Refresh(ctx, optrefresh.ProgressStreams(progressStreams...))
```

Comment:
This should do the same thing, and looks cleaner.
```suggestion
	progressStreams := optrefresh.ProgressStreams(os.Stdout, tmp)
	res, err = s.Refresh(ctx, progressStreams)
```